packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-iam-fields.ts

```diff
@@ -0,0 +1,527 @@
+import {
+  addApiWithAllAuthModes,
+  amplifyPush,
+  createNewProjectDir,
+  deleteDBInstance,
+  deleteProject,
+  deleteProjectDir,
+  enableUserPoolUnauthenticatedAccess,
+  getAppSyncApi,
+  getProjectMeta,
+  importRDSDatabase,
+  initJSProjectWithProfile,
+  setupRDSInstanceAndData,
+  sleep,
+} from 'amplify-category-api-e2e-core';
+import { existsSync, removeSync, writeFileSync } from 'fs-extra';
+import generator from 'generate-password';
+import path from 'path';
+import { GQLQueryHelper } from '../query-utils/gql-helper';
+import { configureAmplify, getConfiguredAppsyncClientIAMAuth, getUserPoolId, setupUser, signInUser } from '../schema-api-directives';
+import { ImportedRDSType } from '@aws-amplify/graphql-transformer-core';
+import { SQL_TESTS_USE_BETA } from './sql-e2e-config';
+import {
+  checkListItemExistence,
+  checkListResponseErrors,
+  checkOperationResult,
+  configureAppSyncClients,
+  expectedFieldErrors,
+  expectedOperationError,
+  getDefaultDatabasePort,
+} from '../rds-v2-test-utils';
+import { Auth } from 'aws-amplify';
+import { schema, sqlCreateStatements } from '../__tests__/auth-test-schemas/userpool-iam-fields';
+
+// to deal with bug in cognito-identity-js
+(global as any).fetch = require('node-fetch');
```

Comment:
nit: This probably could be included in the test environment to avoid repeated copy/paste. https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-e2e-core/src/cli-test-environment.js